src/js/layout.js

```diff
@@ -10,10 +10,11 @@ class Layout {
 		//Default options
 		this.options = Object.assign({}, {
 			baseClass: 'o-layout',
-			constructNav: true
+			constructNav: true,
+			navHeadingSelector: 'h2, h3'
 		}, options || Layout.getDataAttributes(layoutEl));
 
-		this.headings = [...document.querySelectorAll('h2, h3')]
+		this.headings = [...document.querySelectorAll(this.options.navHeadingSelector)]
```

Comment:
Currently this will find all nodes on the page which match the navHeadingSelector, is this intentional? We could grab the o-layout node and then do a scoped querySelectorAll to return only nodes within the o-layout component.
```js
this.layoutEl.querySelectorAll(`:scope ${this.options.navHeadingSelector}`);
```

Reply:
For now I've updated to `this.layoutEl.querySelectorAll`. Adding the scope is more complex as the selector may contain commas so you'd end up with the selector: `.example-scope h2, h3`. As commas can appear in attribute selector values, we'd have to parse the selector and add the scope in front of each part which seems a lot of work :)

Reply:
It would not be `.example-scope h2, h3`, it would be `:scope h2, h3`. No work should be required in combining selectors.

`:scope` is a special selector that informs the user-agent to only look in the node-tree within the current node.